mllib-tests/v1p5/src/main/scala/mllib/perf/MLAlgorithmTests.scala

```diff
@@ -591,28 +623,72 @@ class DecisionTreeTest(sc: SparkContext) extends DecisionTreeTests(sc) {
     if (labelType == 0) {
       // Regression
       ensembleType match {
-        case "RandomForest" =>
-          Left(RandomForest.trainRegressor(rdd, categoricalFeaturesInfo, numTrees, featureSubsetStrategy,
-            "variance", treeDepth, maxBins, this.getRandomSeed))
-        case "GradientBoostedTrees" =>
+        case "mllib.RandomForest" =>
+          MLlibRFModel(RandomForest.trainRegressor(rdd, categoricalFeaturesInfo, numTrees,
+            featureSubsetStrategy, "variance", treeDepth, maxBins, this.getRandomSeed))
+        case "ml.RandomForest" =>
```

Comment:
MLlib algorithms use categoricalFeaturesInfo, which specifies which features are categorical.  That's been replaced with the Attributes.  MLlib trees require communicating more data for continuous features than for categorical (in general), so it's important that MLlib and ML trees treat features in the same way.